packages/core/src/mappings/TextInput.js

```diff
@@ -8,186 +8,263 @@ import {
-  },
-  triggers: [Triggers.OnChangeText],
-  props: {
-    ...TEXT_INPUT_PROPS,
-    style: {
-      group: GROUPS.basic,
-      label: "Style",
-      description: "Text Style",
-      editable: false,
-      required: false,
-      formType: FORM_TYPES.typeStyle,
-      propType: PROP_TYPES.THEME,
-      defaultValue: null,
-    },
-    clearButtonMode: {
-      group: GROUPS.basic,
-      label: "Clear Button Mode",
-      description:
-        "Enables a button within the textInput to clear the data entered",
-      editable: true,
-      required: false,
-      options: ["never", "while-editing", "unless-editing", "always"],
-      defaultValue: null,
-      formType: FORM_TYPES.flatArray,
-      propType: PROP_TYPES.STRING,
-    },
-    clearTextOnFocus: {
-      group: GROUPS.basic,
-      label: "Clear Text on Focus",
-      description:
-        "If true, clears the text field automatically when its focused.",
-      editable: true,
-      required: false,
-      defaultValue: null,
-      formType: FORM_TYPES.boolean,
-      propType: PROP_TYPES.BOOLEAN,
-    },
-    enablesReturnKeyAutomatically: {
-      group: GROUPS.basic,
-      label: "Enables Return Key Automatically",
-      description:
-        "If true, the keyboard disables the return key when there is no text and automatically enables it when there is (Default: false)",
-      editable: true,
-      required: false,
-      defaultValue: null,
-      formType: FORM_TYPES.boolean,
-      propType: PROP_TYPES.BOOLEAN,
-    },
-    multiline: {
-      group: GROUPS.basic,
-      label: "Multiple Lines",
-      description:
-        "Allows multiple lines of input, useful for situations where the user may be typing in a lot of data.",
-      editable: true,
-      required: false,
-      defaultValue: null,
-      formType: FORM_TYPES.boolean,
-      propType: PROP_TYPES.BOOLEAN,
+export const SEED_DATA = [
+  {
+    name: "Text Input",
+    tag: "TextInput",
+    description: "An input field that allows users to type in data.",
+    category: COMPONENT_TYPES.input,
+    preview_image_url:
+      "https://res.cloudinary.com/altos/image/upload/draftbit/Jigsaw/TextInput.png",
+    supports_list_render: false,
+    layout: {
+      borderLeftWidth: 1,
+      borderRightWidth: 1,
+      borderTopWidth: 1,
+      borderBottomWidth: 1,
+      borderColor: "divider",
+      paddingLeft: 8,
+      paddingRight: 8,
+      paddingTop: 8,
+      paddingBottom: 8,
+      borderRadius: 8,
     },
-    numberOfLines: {
-      group: GROUPS.basic,
-      label: "Number of Lines",
-      description:
-        "Sets the number of lines for the input (Multiple Lines needs to be true)",
-      editable: true,
-      required: false,
-      defaultValue: null,
-      min: 0,
-      step: 1,
-      precision: 0,
-      formType: FORM_TYPES.number,
-      propType: PROP_TYPES.NUMBER,
+    triggers: [Triggers.OnChangeText],
+    props: {
+      ...TEXT_INPUT_PROPS,
+      style: {
```

Comment:
If these other props, that aren't a part of `TEXT_INPUT_PROPS`, are all shared between both the TextInput and the new Number Input, could you put them in an object above the seed data and then use them in the seed data?

So like:

```js
const NATIVE_INPUT_PROPS = {
  style: {
    whatever
  },
  clearButtonMode: {
    whatever
  }, etc...
};

export const SEED_DATA = [{
  name: "Text Input",
  tag: "TextInput",
  props: {
    ...TEXT_INPUT_PROPS,
    ...NATIVE_INPUT_PROPS,
  }
}, {
  name: "Number Input",
  tag: "NumberInput",
  props: {
    ...TEXT_INPUT_PROPS,
    ...NATIVE_INPUT_PROPS,
  }
];
```

Reply:
updated